### PR TITLE
28 implement UI for landing page categories section

### LINKED ISSUE #28: Implement UI for Landing Page categories section
# Implement UI Section for Landing Page

## Description
We need to create a UI section for the landing page using React, TailwindCSS, and Redux. This section includes a header, sidebar with categories, and a main content area displaying products. The goal is to develop a responsive and styled user interface according to the provided design.

## Tasks

1. **Basic Components**:
   

 - [ ]  Create a `ProductCard` component to display individual product details.

 - [ ] Create a `ProductsList` component to map over product data and render multiple `ProductCard` components.

2. **Main Layout**:
 - [ ] Create a `MainLayout` component to structure the overall page.
 - [ ] Divide the layout into sidebar and main content areas.

3. **Categories Component**:
 - [ ] Create a `Categories` component to display a list of product categories.
 - [ ] Integrate the `Categories` component into the sidebar of the `MainLayout`.

4. **Header Component**:
 - [ ] Create a simple header component with static information such as delivery, support, and discounts.

5. **Connect Components**:
 - [ ] Integrate `ProductsList` into the main content area of `MainLayout`.
 - [ ] Ensure components are styled and aligned according to the design.

6. **Dynamic Data**:
 - [ ] Fetch product data from the Redux store in the `ProductsList` component.
 - [ ] Ensure `ProductCard` displays data dynamically based on the Redux state.

7. **Styling and Responsiveness**:
 - [ ] Apply TailwindCSS classes to ensure the design matches the provided image.
 - [ ] Ensure the layout is responsive and looks good on different screen sizes.

8. **Code Review and Testing**:
  - [ ] Review code for best practices and readability.
    [ ] Test the application to ensure all components render correctly and interact as expected.

## Acceptance Criteria
- The header component displays static information correctly.
- The sidebar component lists all product categories.
- The main content area correctly displays a list of products with the specified details.
- Components fetch and display data dynamically from the Redux store.
- The UI is styled according to the provided design and is fully responsive.
- All code is reviewed, tested, and documented.

## Definition of Done
- The React application is fully functional and styled using TailwindCSS.
- All components are integrated with Redux for state management.
- The UI matches the provided design and is responsive across different screen sizes.
- The codebase is reviewed for best practices, readability, and is thoroughly tested.
- Documentation is complete, including project setup instructions and usage guidelines.
- The application is ready for deployment and use by the end-users.

### PR DESCRIPTION
**Pull request by niyobern**
## Pull Request Title: Implement Landing Page UI with Product Display

This pull request introduces the initial UI structure for our landing page, Categories Section. The core components have been developed using React, styled with TailwindCSS, and integrated with Redux for state management.

### **Key Changes:**

*   **Components:**
    *   `ProductCard`: Displays individual product information (image, title, price).
    *   `ProductsList`: Renders a dynamic list of `ProductCard` components based on Redux data.
    *   `Sidebar`: Lists available product categories in the sidebar or as a dropdown on mobile.
    *   `Header`: Presents static information (delivery, support, discounts).
    *   `CategoriesSection`: Structures the page with sidebar and main content areas.

*   **Redux Integration:**
    *   Product data is fetched from the Redux store in `ProductsList`.
    *   `ProductCard` renders dynamically based on the Redux state.

*   **Styling:**
    *   TailwindCSS classes applied for visual consistency with the provided design.
    *   Responsive design ensures a seamless experience across devices.

*   **Testing:**
    *   Thorough component testing to verify rendering and data flow.

### **Screenshots:**
 
**Tests Screenshot**
image

**UI Mobile**
image

**UI Desktop**
image

### **How to Review:**

1.  Checkout this branch and run the application.
2.  Verify that the landing page matches the provided design.
3.  Test product display on different screen sizes (responsiveness).
4.  Check that product data is fetched correctly from the Redux store.
5.  Review code for clarity, best practices, and adherence to project guidelines.

### **Next Steps:**

Once this PR is merged, we can focus on enhancing the UI with additional features and refining the overall user experience.

Let me know if you have any other questions or requests!

#28

### CODE REVIEW

**Review by 13XAVI — APPROVED** [review]

**Review by 13XAVI — CHANGES_REQUESTED** [review]
Reduce commit message minimum 2

**Review by ambroisegithub — APPROVED** [review]
It looks good for merge!!!

**Review by 13XAVI — APPROVED** [review]

**Review by Dawaic6 — APPROVED** [review]

File: src/app/slices/categorySlice.ts [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+
+interface Payload {
+  message: string;
+  data: Category[];
+}
+export const fetchCategories = createAsyncThunk(
+  'categories/fetchCategories',
+  async () => {
+    const response = await axios.get<Payload>(
+      'https://dynamites-ecomm-be.onrender.com/api/v1/category/'
```
Comment (dushimeemma) [inline src/app/slices/categorySlice.ts]: this should be in environment variables

File: src/components/home/categories.tsx [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+                    />
+                  </g>
+                </svg>
+                <input
+                  placeholder="Search"
+                  type="text"
+                  className="w-full outline-none bg-gray-100 placeholder:text-gray-400 font-light"
+                />
+              </div>
+              <div className="bg-primary px-8 rounded text-gray-200 flex text-center items-center">
```
Comment (dushimeemma) [inline src/components/home/categories.tsx]: reuse the button from the hero section

File: src/components/home/header.tsx [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+      className="bg-white py-4 mx-0 md:mx-16"
+      style={{
+        fontFamily:
+          'Lexend, Manrope, Poppins, system-ui, -apple-system, Segoe UI, Roboto, Helvetica Neue, Arial, Noto Sans, Liberation Sans, sans-serif',
+      }}
+    >
+      <div className="mx-auto p-4 flex flex-col md:flex-row justify-between gap-4 md:items-center">
+        <div className="flex items-center gap-2">
+          <img src="/src/assets/icons/icon1.svg" alt="Shipping Icon" />
+          <div className="flex flex-col gap-2">
```
Comment (dushimeemma) [inline src/components/home/header.tsx]: make it a reusable component, and I would suggest to make the List of objects that include image, title and description and loop through it.

**Review by dushimeemma — CHANGES_REQUESTED** [review]

File: src/components/home/header.tsx (hunk 3 above)
Comment (niyobern) [reply src/components/home/header.tsx]: done

File: src/components/home/categories.tsx (hunk 2 above)
Comment (niyobern) [reply src/components/home/categories.tsx]: done

**Pull request merged**